docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+# Design for configuring the SDK
+
+This document outlines some of the goals we have for user configuration of the SDK. It is a
+continuation of discussion started with https://github.com/open-telemetry/opentelemetry-java/issues/2022.
+
+## Target audiences
+
+There are a few different target audiences that are related to our configuration story.
+
+- Application developers, aka end-users. Often have no knowledge of tracing but want to add 
+OpenTelemetry to their app and see traces show up in a console. Application developers will increase
+in number forever, while the below are more constant.
+
+- Dev-ops / framework developers. Write components or frameworks to support providing tracing to
+their application developers. May write custom SDK extensions such as exporters, samplers, to fit
+with their internal infrastructure and as such have some familiarity with tracing at least as the
+SDK presents it.
+
+- OpenTelemetry maintainers. Write the SDK code.
+
+When making decisions, especially about complexity, we always prioritize the application developers,
+then framework developers, then maintainers. This is because we expect those with less domain
```

Comment:
this is an excellent call-out, and something we all definitely need to keep in mind. Sometimes this will mean we add "more than one way to do things" because it will make app developers lives easier, even if it makes our jobs as maintainers more annoying/painful.